spec/eof.md

```diff
@@ -103,6 +103,47 @@ On top of the types defined in the table above, the following validity constrain
 - the total size of not yet deployed container might be up to `data_size` lower than the above values due to how the data section is rewritten and resized during deployment (see [Data Section Lifecycle](#data-section-lifecycle))
 - the total size of a container must not exceed `MAX_INITCODE_SIZE` (as defined in EIP-3860)
 
+## Transaction Types
+
+Introduce new transaction type `InitcodeTransaction` which extends EIP-1559 (type 2) transaction by adding a new field `initcodes: List[ByteList[MAX_INITCODE_SIZE], MAX_INITCODE_COUNT]`.
+
+The `initcodes` can only be accessed via the `TXCREATE` instruction (see below), therefore `InitcodeTransactions` are intended to be sent to contracts including `TXCREATE` in their execution.
+
+We introduce a standardised Creator Contract (i.e. written in EVM, but existing at a known address, such as precompiles), which eliminates the need to have create transactions with empty `to`. The Creator Contract will be predeployed at the EOF activation block. Note that such introduction of the Creator Contract is needed, because only EOF contracts can create EOF contracts. See the appendix below for Creator Contract code.
+
+Under transaction validation rules `initcodes` are not validated for conforming to the EOF specification. They are only validated when accessed via `TXCREATE`. This avoids potential DoS attacks of the mempool. If during the execution of an `InitcodeTransaction` no `TXCREATE` instruction is called, such transaction is still valid.
+
+`initcodes` data is similar to calldata for two reasons:
+1) It must be fully transmitted in the transaction.
+2) It is accessible to the EVM, but it can't be fully loaded into EVM memory.
+
+For these reasons, define cost of each of the `initcodes` items same as calldata (16 gas for non-zero bytes, 4 for zero bytes -- see EIP-2028). The intrinsic gas of an `InitcodeTransaction` is extended by the sum of all those items' costs.
```

Comment:
I wonder how this interacts with EIP-7623, probably in terms of it `tokens_in_calldata` should be extended with counting the tokens of `initcodes` (which in the end is the same as extending intrinsic gas with `initcodes` cost and applying calldata cost floor)

Reply:
yea, good catch. I think the intent is that initcodes is charged for just like regular calldata. I propose we delay speccing this, just in case 7623 undergoes some last minute change. I'll leave an issue to handle this

Reply:
It's contract data.  Perhaps we just outright charge the higher rate, and say that encompasses the INITCODE_WORD_COST as well as all EOF code validation costs.

Reply:
Floor rate is pretty high, 2.5 times more expensive than regular. I've updated this paragraph to be based on EIP-7623 and initcodes price is the same as calldata for now.

Reply:
Thank you. Having the schedule as in EIP-7623 I think we can resolve this conversation and also resolve the issue to handle this #178 ?